### PR TITLE
Add benchmark tests for TrieDB

### PR DESCRIPTION
**Pull request by Remagpie**
Resolve https://github.com/CodeChain-io/codechain/issues/1502

You can run the benches by running following commands under `util/merkle` directory:
```
cargo +nightly bench
```

~Here's the actual benchmark statistics I got from the real machines.~
See https://github.com/CodeChain-io/codechain/pull/1828#issuecomment-546818670 for the new result.

~with SSD:~
```
test bench_read_multiple  ... bench: 239,979,231 ns/iter (+/- 98,801,879)
test bench_read_single    ... bench:      31,041 ns/iter (+/- 2,420)
test bench_write_multiple ... bench: 219,346,906 ns/iter (+/- 123,023,116)
test bench_write_single   ... bench:       6,788 ns/iter (+/- 1,069)
```

~with HDD:~
```
test bench_read_multiple  ... bench: 228,591,915 ns/iter (+/- 1,163,440)
test bench_read_single    ... bench:      67,821 ns/iter (+/- 1,498)
test bench_write_multiple ... bench: 534,238,617 ns/iter (+/- 2,717,086,442)
test bench_write_single   ... bench:      43,088 ns/iter (+/- 5,151)
```

~Write is faster(!) than read in SSD. I'm checking if something went wrong now.~

### CODE REVIEW

File: util/merkle/benches/triedb.rs [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+#[bench]
+fn bench_read_single(b: &mut Bencher) {
+    let mut db = TestDB::new(DB_SIZE);
+    b.iter(|| {
+        let trie = db.trie();
+        let item = random::<usize>() % DB_SIZE;
+        let bytes = trie.get(&item.to_be_bytes()).unwrap().unwrap();
+        assert_eq!(item, usize::from_be_bytes((*bytes).try_into().unwrap()));
+        db.flush();
```
Comment (sgkim126) [inline util/merkle/benches/triedb.rs]: Any reason to call flush to test reading?

Comment (Remagpie) [reply util/merkle/benches/triedb.rs]: No specific reason for that. It was just one of my trial-an-errors.
I think we don't need it though. Do you agree with that?

Comment (sgkim126) [reply util/merkle/benches/triedb.rs]: Yes, I think we don't need it to test reading.

**Comment by Remagpie** [comment]
Benchmarks for reads got faster after removing `db.flush()`.
Now the statistics are similar to the numbers I expected.
Here are the new results:

Used machine: [n1-highcpu-2 in Google Cloud Platform](https://cloud.google.com/compute/docs/machine-types#n1_machine_types)

with SSD:
```
test bench_read_multiple  ... bench: 207,037,788 ns/iter (+/- 4,241,485)
test bench_read_single    ... bench:      21,873 ns/iter (+/- 1,422)
test bench_write_multiple ... bench: 402,642,290 ns/iter (+/- 146,816,535)
test bench_write_single   ... bench:       7,414 ns/iter (+/- 2,509)
```

with HDD:
```
test bench_read_multiple  ... bench: 211,523,461 ns/iter (+/- 6,263,803)
test bench_read_single    ... bench:      22,938 ns/iter (+/- 1,260)
test bench_write_multiple ... bench: 753,595,731 ns/iter (+/- 2,197,507,866)
test bench_write_single   ... bench:       6,989 ns/iter (+/- 2,760)
```

**Pull request merged**